### PR TITLE
fix: travis and clippy

### LINKED ISSUE #11: Remove old nightly from Travis
https://github.com/filecoin-project/merkle_light/blob/63a609decf3cc552b0aa79477b828f4a493f725d/.travis.yml#L12

Not sure what's the clean way to do this as Clippy might depend on it:

https://github.com/filecoin-project/merkle_light/blob/63a609decf3cc552b0aa79477b828f4a493f725d/.travis.yml#L21-L23

It's the only version that is failing in the tests:

https://github.com/filecoin-project/merkle_light/pull/10/checks?check_run_id=71316907

### PR DESCRIPTION
**Pull request by dignifiedquire**
Closes #11

### CODE REVIEW

File: merkle/src/hash.rs [hunk 1]
```diff
@@ -116,7 +116,6 @@ where
     T: Clone + AsRef<[u8]>,
 {
     /// Returns the hash value for the data stream written so far.
-    #[inline]
```
Comment (schomatis) [inline merkle/src/hash.rs]: Why removing this?

Comment (dignifiedquire) [reply merkle/src/hash.rs]: Because `inline` only has an effect on functions with a body, and clippy reminded me to fix it

**Comment by schomatis** [comment]
(Note: is fine that test keep failing, see https://github.com/filecoin-project/merkle_light/pull/10.)

**Comment by dignifiedquire** [comment]
yeah just need to get clippy working again

**Pull request merged**